### PR TITLE
Enable more hw intrinsics for AOT

### PR DESCRIPTION
**Pull request by EgorBo**
Enable all `System.Numerics.Vector<T>` intrinsics in AOT mode (for netcore). Almost all of them don't require any CPU feature check on x86 (SSE1-2). Some require SSE4x (e.g. `get_Item`) but those are LLVM-only. LLVM is able to workaround lack of SSE4x for them (e.g. via sse2's movemask in this case)

Enable `Math(F).Round` if SSE4.1 is presented, e.g. `--aot=mattr=+sse4.1` or `--aot=llvm,mcpu=native` (fixes https://github.com/mono/mono/issues/16178).

### CODE REVIEW

File: mono/mini/aot-compiler.c [hunk 1]
```diff
@@ -8002,7 +8002,7 @@ parse_cpu_features (const gchar *attr)
 	// TODO: neon, sha1, sha2, asimd, etc...
 #endif
 
-	if (!enabled)
+	if (enabled)
```
Comment (EgorBo) [inline mono/mini/aot-compiler.c]: last-minute typo in my previous PR 🙁

**Comment by EgorBo** [comment]
@monojenkins build failed

File: mono/mini/mini.h [hunk 2]
```diff
@@ -2846,6 +2846,8 @@ typedef enum {
 #endif
 } MonoCPUFeatures;
 
+MonoCPUFeatures get_cpu_features (MonoCompile* cfg);
+
```
Comment (vargaz) [inline mono/mini/mini.h]: This should have a mini_ prefix.

**Comment by vargaz** [comment]
Also, this lane failed:
https://jenkins.mono-project.com/job/test-mono-pull-request-amd64-prefix/5913/
The failure looks relevant.

**Comment by EgorBo** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
One lane fails with:
/mnt/jenkins/workspace/test-mono-pull-request-amd64-prefix/mono/mini/mini.c:4313: undefined reference to `mono_llvm_get_cpu_features'

**Comment by vargaz** [comment]
@monojenkins build failed

File: mono/mini/mini.c [hunk 3]
```diff
@@ -4301,3 +4301,25 @@ mono_target_pagesize (void)
...
+MonoCPUFeatures
+mini_get_cpu_features (MonoCompile* cfg)
+{
+	MonoCPUFeatures features = (MonoCPUFeatures)0;
+#if !defined(MONO_CROSS_COMPILE)
+	if (!cfg->compile_aot || cfg->use_current_cpu) {
+		// detect current CPU features if we are in JIT mode or AOT with use_current_cpu flag.
+#if defined(ENABLE_LLVM) && !defined(MONO_LLVM_LOADED)
+		features = mono_llvm_get_cpu_features (); // llvm has a nice built-in API to detect features
+#elif defined(TARGET_AMD64)
```
Comment (vargaz) [inline mono/mini/mini.c]: This should be fixed in mini-llvm-loaded.c.

**Comment by vargaz** [comment]
@monojenkins build failed

File: mono/mini/mini-llvm-loaded.c [hunk 4]
```diff
@@ -22,6 +22,7 @@ typedef struct {
 	void (*emit_aot_data)(const char *symbol, guint8 *data, int data_len);
 	void (*free_domain_info)(MonoDomain *domain);
 	void (*create_vars)(MonoCompile *cfg);
+	MonoCPUFeatures (*mono_llvm_get_cpu_features)(void);
```
Comment (filipnavara) [inline mono/mini/mini-llvm-loaded.c]: Strip `mono_llvm` here.

File: mono/mini/mini-llvm-loaded.c [hunk 5]
```diff
@@ -143,6 +144,11 @@ mono_llvm_load (const char* bpath)
 	return FALSE;
 }
 
+MonoCPUFeatures mono_llvm_get_cpu_features (void)
+{
+	return backend.mono_llvm_get_cpu_features (cfg);
```
Comment (filipnavara) [inline mono/mini/mini-llvm-loaded.c]: You need to add it to `mono_llvm_load` above.

**Comment by EgorBo** [comment]
@monojenkins build failed

**Comment by EgorBo** [comment]
@monojenkins build failed

**Comment by filipnavara** [comment]
@monojenkins build Linux ARMv7

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**